### ISSUE TITLE
Quoting fails due to quoted amount being outside of curve

### ISSUE THREAD

**Issue by dappelt** [interledgerjs/ilp-routing#43]
**TL;DR** I encountered a quote failure that seems to be related to the recent currency scale change. It looks like the problem is that the liquidity curve is not scaled in the same fashion as the quoted amounts. Therefore, the quoted amount is outside of the liquidity curve and the quote fails, even for small amounts.

Several modules are invovled; let me walk you through an example to figure out which one needs fixing:

`ILP.SPSP.quote(...)` is called with `destinationAmount: 0.01`, which results in the following quote request from `ILQP`:
```
ilp:ilqp quoting 10000000000000 (destination amount) to g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q via [ 'g.test.ledger1.connie' ] +0ms
ilp:ilqp remote quote connector=g.test.ledger1.connie query={"source_address":"g.test.ledger1.alice3","destination_address":"g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q","destination_amount":"10000000000000"}
```

Note that `ILQP` is translating `destinationAmount: 0.01` into the integer `10000000000000` (all involved ledgers use the default for currency scale). This is the log output of the connecotr receiving the quote request:

```
2017-05-18T09:16:03.253Z ilp-core debug got incoming message: {"ledger":"g.test.ledger1.","account":"g.test.ledger1.alice3","from":"g.test.ledger1.alice3","to":"g.test.ledger1.connie","data":{"method":"quote_request","data":{"source_address":"g.test.ledger1.alice3","destination_address":"g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q","destination_amount":"10000000000000"},"id":"6a730c77-2f98-4dd4-bc71-14de64f70486"}}
2017-05-18T09:16:03.255Z connector:route-builder info creating quote sourceAddress=g.test.ledger1.alice3 sourceAmount=undefined destinationAddress=g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q destinationAmount=10000000000000 slippage=undefined
2017-05-18T09:16:03.255Z ilp-routing:routing-tables debug searching best hop from g.test.ledger1.alice3 to g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q for 10000000000000 (by dst amount)
2017-05-18T09:16:03.256Z ilp-routing:routing-table debug findBestHopForDestinationAmount could not find route to g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q for 10000000000000. Current routing table: {"g.test.ledger2.":{}}
2017-05-18T09:16:03.256Z connector:route-builder info no quote found for params: {"sourceAddress":"g.test.ledger1.alice3","destinationAddress":"g.test.ledger2.bob3.x2_z-1Kc04c5Ig8AvfKJVvX4U9_NNJ50Q","destinationAmount":"10000000000000","explain":false}
```

As you can see from the log, the connector fails to provide a quote. A look at the routing table makes it apparent why the quote is failing:

```
"source_ledger": "g.test.ledger2.", "destination_ledger": "g.test.ledger1.", "points": [
    [1.002004008016032e-15, 0],
    [1000000000000, 998000000000]
], "min_message_window": 1, "added_during_epoch": 0, "paths": [
    []
], "source_account": "g.test.ledger2.connie"
}, {
    "source_ledger": "g.test.ledger1.",
    "destination_ledger": "g.test.ledger2.",
    "points": [
        [1.002004008016032e-15, 0],
        [1000000000000, 998000000000]
    ],
    "min_message_window": 1,
    "added_during_epoch": 0,
    "paths": [
        []
    ],
    "source_account": "g.test.ledger1.connie"
}
```

The quoted destination amount (10.000.000.000.000) is outside of the liquidity curve (the curve ends at [1.000.000.000.000, 998.000.000.000]) and, hence, the quote fails. Note that the connector has sufficient funds to deliver the quoted amount of `0.01`.

**Comment by michielbdejong** [comment]
Ha, good catch!

I already anticipated this a bit and added some zeroes in https://github.com/interledgerjs/ilp-connector/pull/337/files#diff-732a4256af7e30fb991b56df0f8474f2 but will add some more! :)

**Comment by dappelt** [comment]
How about letting the liquidity curve end at a point that actually represents the funds a connector has available? Also, for that type of failure, the connector should probably not respond with `... could not find route ...`, but with `Liquidity Exhausted`.

**Comment by michielbdejong** [comment]
yeah, I wanted to touch the code as little as possible. I'll set it to `Math.pow(10, 100)`.

**Comment by michielbdejong** [comment]
I think this is fixed now
